app/src/main/java/com/example/myplaylistmaker/SearchHistory.kt

```diff
@@ -0,0 +1,57 @@
+import android.content.SharedPreferences
+import android.widget.Toast
+import com.example.myplaylistmaker.TRACK_HISTORY_KEY
+import com.example.myplaylistmaker.Track
+import com.example.myplaylistmaker.TrackAdapter
+import com.google.gson.Gson
+
+class SearchHistory(private val sharedPrefs: SharedPreferences) {
+
+    private val gson = Gson()
+
+    private fun createJsonFromTrack(tracks: ArrayList<Track>): String {
+        return gson.toJson(tracks)
+    }
+
+    private fun createTracksFromJson(tracks: String?): ArrayList<Track> {
+        return if (tracks != null) {
+            val list = gson.fromJson(tracks, Array<Track>::class.java).toList()
+            ArrayList(list)
+        } else {
+            ArrayList()
+        }
```

Comment:
🍏  Можно написать в стиле kotlin:
```suggestion
        return tracks?.let { trackList ->
            val array = gson.fromJson(trackList, Array<Track>::class.java)
            arrayListOf(*array)
        } ?: arrayListOf()
```
Дополнительно рекомендую почитать про scope functions и работу с элементами массива:
https://kotlinlang.org/docs/arrays.html#pass-variable-number-of-arguments-to-a-function
https://kotlinlang.org/docs/scope-functions.html